### ISSUE TITLE
observable does not define property since 1.0.5

### ISSUE THREAD

**Issue by Thanood** [aurelia/binding#511]
**I'm submitting a bug report**

* **Library Version:**
1.0.5

**Please tell us about your environment:**
* **Operating System:**
Windows 8

* **Node Version:**
6.2.2

* **NPM Version:**
3.10.5

* **JSPM OR Webpack AND Version**
JSPM 0.16.15

* **Browser:**
all

* **Language:**
all

**Current behavior:**

`observable` does not define the given property on the target object unless it's a class decorator.
This behaviour was introduced in https://github.com/aurelia/binding/commit/6586cd2 (line 52).

**Expected/desired behavior:**

* **What is the expected behavior?**

If not present, the property is defined in the target object.

* **What is the motivation / use case for changing the behavior?**

I have to use ES5 in parts of my current app. I'm defining observables like so:

```javascript
var fw = require('aurelia-framework');
function MyClass() {
  fw.observable(this, 'value');
}
MyClass.prototype.attached() {
  this.value = false;
}
MyClass.prototype.valueChanged(newValue) {
  fireEvent(this.checkbox, 'change', { value: newValue });
}
```

Because the `value` property is not defined in the constructor (by `observable`) with a setter decoration, it is defined (overwritten) on attached. The `valueChanged` function is never called. This broke all of my filters after updating from v1.0.4. :smiley:

Edit: it doesn't help to define `value` before calling `observable`.
Downgrading to 1.0.4 fixed this for me.

**Comment by estaphorsius** [comment]
I'm having the same issue over here. We are using aurelia with TypeScript. Building our application results in a compilation error:

Error: typings/modules/aurelia-binding/index.d.ts(193,14): error TS2420: Class 'DataAttributeObserver' incorrectly implements interface 'InternalPropertyObserver'.
  Property 'getValue' is missing in type 'DataAttributeObserver'.

It seems typing for DataAttributeObserver is missing a definition.

**Comment by fracz** [comment]
I can confirm it is fixed in 1.0.6. Thank you!

**Comment by Gheoan** [comment]
@estaphorsius, @fracz I think this is a different issue.
@Thanood, it works for me: https://gist.run/?id=b1f2df7ae68e97d47bb3a7e7d696cebe.

**Comment by Thanood** [comment]
@Gheoan Depends on the version used in the gist's bundle.
I tried a search for `isClassDecorator` on the bundle and it had no search results. That could mean it's still at 1.0.4.

**Comment by jdanyow** [comment]
@Thanood / @Gheoan I've updated the gist to use the latest aurelia: https://gist.run/?id=01e2d581154c8b3a2c12df2d3b7c214e

**Comment by Thanood** [comment]
Okay, I did not check `hasOwnProperty`, good point. 😃 
@jdanyow Do you get the second alert? I don't.

So I was wrong with my assumption but why does `valueChanged` not fire?

**Comment by Gheoan** [comment]
@Thanood, the correct way to use decorators in es5 is using the `decorators` function:
```js
fw.decorators(fw.observable).on(this, 'value');
```

Here is a gist using decorators function: https://gist.run/?id=72b9f9087a01800c4bc871a90aa5b3b1.
However there is still a bug with observable that prevents the above from working. Will submit a PR for that.

**Comment by Thanood** [comment]
I see, thanks! Will try that tomorrow in my app. I need some sleep now. 😃 
Thank you very much.

**Comment by Gheoan** [comment]
Your assumption was right, in the gist hasOwnProperty('value') is true because value was used on the view and that caused the property to be observed which in fact creates the property.

**Comment by Gheoan** [comment]
Actually, the corect way to use decorators in ES5 is 
```js
fw.decorators(fw.observable).on(MyClass.prototype, 'value');
```

**Comment by Thanood** [comment]
@Gheoan I'm not sure if this happens because it's not implemented (in 1.0.4) or if I did it wrong but I'm getting this error when I try to use decorators in ES5 correctly - on the class prototype:

```
Error: Invalid property descriptor. Cannot both specify accessors and a value or writable attribute
```

This is what I used:

```javascript
fw.decorators(fw.observable).on(TypesViewModel.prototype, 'activeType');
```

I *guess* this is because 1.0.4 uses `Reflect.defineProperty` so this is only for information, hoping that it may help somehow. :smiley: 

I'll change it to use the incorrect approach in the meantime.
Thanks again.
